src/lib/gravityErrorHandler.ts

```diff
@@ -36,6 +36,27 @@ export const GravityMutationErrorType = new GraphQLObjectType<
   }),
 })
 
+export const formatGravityHttpError = (error) => {
```

Comment:
I am replying to both ([1](https://github.com/artsy/metaphysics/pull/5076#pullrequestreview-1475614305), [2](https://github.com/artsy/metaphysics/pull/5076#discussion_r1227114160)) comments you've made here:

> I think rather than add a completely new method for parsing Gravity errors, and using it in these two mutations, we should try to consolidate, and move that logic into the existing parser for Gravity errors, thus allowing it to handle more formats.

100% agree, there is indeed an opportunity to unify the behaviour. I would love to do so, but not in scope of my ticket though.

> Rather than a new method, formatGravityHttpError (what's the significance of 'http' in the naming?) - could this be rolled into the below formatGravityError?

What I've tried to achieve in this PR is to fix an error for these 2 particular mutations. There is a launch blocking bug in Eigen ([link](https://www.notion.so/artsy/Broken-validation-message-when-renaming-an-artwork-list-64758552140c42bd9170d49bb79cc143?pvs=4)), and without this change I cannot proceed with a fix.

Initially I've tried to fix `formatGravityError` instead of introducing a new one, but then decided to avoid changing the method on which rely a bunch of other mutations. I will quote last sentences from the PR description

> First I've tried to fix / refactor existing implementation, but it appeared to be harder (also I bit scary) than expected. A lot of mutations depend on this function and I'm afraid to change it's behaviour, what if sudden "fix" of it will expose some error messages which are not meant to be displayed on frontend 🤷 But I think I'm overthinking. I've decided to extract behaviour I need into a new function and refactor later, but if you think that it's better to refactor right away - let me know. I will be happy to pair on it!

Reply:
I agree that the existing formatting gravity code in MP is pretty abysmal. There's tech debt in this area, which is why this functionality wasn't working and why its not even easy to update the functionality.

For instance, we re-throw as an `HTTPError`, and put `request.body` into a string field for the re-thrown error - https://github.com/artsy/metaphysics/blob/eb9a1362323222a5bd22089c4b798f1442af4a89/src/lib/apis/fetch.ts#L83 , that leads to `[Object object]`.

Then, if we're throwing `HTTPError`, in the current method, what's all the code doing after the conditional of `err instanceof HTTPError` -> isn't that always going to be true?

And so forth.

That's the perennial debate w/ tech debt - when to address it, should we just fulfill a ticket and get out of there, and so forth.

I pushed a commit to another branch of the kind of update/simplification that would be nice to do, and it has to rely on tests within Metaphysics (that of course stub Gravity error responses...) to verify - https://github.com/artsy/metaphysics/commit/485e086e6f699fdfbc9e2662aa20e04fa2825304

cc @damassi curious your opinion on the tech-debt angle

I'm in tentative agreement to not do the bigger refactor now (but is this tracked somewhere? we really should address it), but I don't know if adding a new method named `formatGravityHttpError` is necessarily the best way - could this be done locally in the mutations that currently need them?

Reply:
> I'm in tentative agreement to not do the bigger refactor now (but is this tracked somewhere? we really should address it)

Sure, I've just created a [ticket in Jira](https://artsyproduct.atlassian.net/browse/FX-4815). I will work on it later in case this PR will be merged

Reply:
I don't want to block work, but I do agree with @mzikherman's points up above. For example: when, from the perspective of MP pinging gravity for data, is an error _not_ an http error of some kind? In each case we're making a request. 

The first check in `formatGravityError` is:

```js
if (error instanceof HTTPError) 
```

But here we're redefining things a bit. 

I understand the idea is to fix the issue and that is def 👍 , but we should avoid file-and-forget it fixes that are legitimately ambiguous for future devs, as is the case here.

Reply:
One suggestion / compromise in terms of unifying `formatGravityHttpError` and `formatGravityError` would be to update the function signature of `formatGravityError` to be partially-applied and thus configurable. Meaning:

```tsx
export const setupGravityError = options => {
  return (error) => {
    ...
  }
}
```
Then one could invoke it like so:

```ts
const formatGravityError = setupGravityError({ verboseHTTP: true })

try {
  ...
} catch (error) {
  const formatGravityError = setupGravityError({ verboseHTTP: true })
  return formatGravityError(error)
}
```